.github/workflows/release.yaml

```diff
@@ -52,13 +52,13 @@ jobs:
 
       - name: Create and switch to dedicated release branch
         run: |
-          git checkout -b release-${{ github.event.inputs.releaseVersion }}
+          git checkout -b "release-${{ github.event.inputs.releaseVersion }}"
 
       - name: Release
-        run: |
-          mvn --batch-mode -s settings.xml release:clean release:prepare -Dtag=v${{ github.event.inputs.releaseVersion }} -DreleaseVersion=${{ github.event.inputs.releaseVersion }} -DdevelopmentVersion=${{ github.event.inputs.developmentVersion }}-SNAPSHOT -DscmReleaseCommitComment="[release] Set release & tag: ${{ github.event.inputs.releaseVersion }}" -DscmDevelopmentCommitComment="[release] Set next version: ${{ github.event.inputs.developmentVersion }}-SNAPSHOT"
-          mvn --batch-mode -s settings.xml release:perform -Dusername=${{ env.GITHUB_USERNAME }} -Dpassword=${{ env.GITHUB_TOKEN }} -DskipTests=true
+        run: scripts/release.sh
```

Comment:
It could but when it is time to sending credentials to an external plugin I'm in zero trust mood

PROS for home made:
- home made recipes not sending credentials (`mvn release` is itself a mvn plugin, but why to exposing credentials twice if only one is enough and [sources are public](https://github.com/apache/maven-release/) )
- full control onto our script if fixes or enhancements are required

CONS for external action:
- tie to a 3rd party plugin 
- what if maintainer do not respond/provide support
- what if plugin is no longer maintained

Reply:
In my test I like not to reinvent the wheel but it's up to.